### ISSUE TITLE
Untitled story 6342

### ISSUE THREAD

**Issue by mmuller** [instedd/cdx#1445]

**Comment by devduarte** [comment]
@mmuller @diegoliberman What would be the expected result for an unathorized action? A notification telling the user that is not authorized for the given action? redirect it to some kind of a "forbidden page" ?

**Comment by diegoliberman** [comment]
> @mmuller @diegoliberman What would be the expected result for an unathorized action? A notification telling the user that is not authorized for the given action? redirect it to some kind of a "forbidden page" ?

I would check how it behaves in other entities, such as patients.  I'll check and add the expected behaviour here.  Anyhow, letting the 403 default error appear, is not a nice UX.

**Comment by devduarte** [comment]
@diegoliberman We are actually re-using the same behaviour that's been used in other entities. Maybe this is a wider issue that would involve solving the 403 status response in general.

**Comment by mmuller** [comment]
@devduarte I think the delete icon should be hidden for institution readers.

**Comment by diegoliberman** [comment]
The behaviour for Patients, is the following:

1. The delete button appears when Editing the patient record.
2. As the user with reader access can't edit the record, they can't see the delete button. 

I tried for Sites and Devices, and they worked like patients. 

Actually, the same is happening for Samples & Batches, in regards to the delete button that appears when viewing the individual Sample/Batch.

The particularity for Samples and Batches, is that it has a 2nd delete action located in the list of records.  This doesn't happen for the rest of the entities.  I guess we did that because it was requested to be able to delete multiple records at once.  

In conclusion, I think the most consistent approach would be to hide the delete button when users don't have enough access. 

![image](https://user-images.githubusercontent.com/25181052/152219071-5708e948-9e77-4355-afb6-5bc2ee6e0dbf.png)

**Comment by mmuller** [comment]
> In conclusion, I think the most consistent approach would be to hide the delete button when users don't have enough access.

@diegoliberman I would say to hide:

1. the delete button in the sample/batch view
2. the delete icon in the samples/batches listing

Do you agree?

**Comment by ysbaddaden** [comment]
Yes, if you aren't authorized to execute an action you shouldn't be presented with the option :)

**Comment by diegoliberman** [comment]
We need to check if the Sample list can contain samples that the user has just read access to some, and write access to others.

**Comment by diegoliberman** [comment]
On hold until we fully understand the permissions in CDx
